### PR TITLE
Deletes Dofs on Boundary for Div-Conforming Case

### PR DESCRIPTION
**Pull request by mx-nlte**

### CODE REVIEW

File: Bembel/src/AnsatzSpace/Glue.hpp [hunk 1]
```diff
@@ -122,17 +124,24 @@ class Glue {
...
 
-    const int post_dofs = pre_dofs - number_of_slaves;
-
+    const int post_dofs =
+        pre_dofs - number_of_slaves - number_of_boundary_dofs;
+    const int glued_dofs = pre_dofs - number_of_slaves;
+    assert(post_dofs != 0 && "All degrees of freedom are on the boundary!");
     // This block is the heart of the algorithms. Skip keeps track of how many
     // slaves have been skipped already, and master_index keeps track on which
     // master is about to be glued next.
```
Comment (flx-wlf) [inline Bembel/src/AnsatzSpace/Glue.hpp:133]: post_index missing in comment

File: Bembel/src/AnsatzSpace/Glue.hpp [hunk 2]
```diff
@@ -122,17 +124,24 @@ class Glue {
...
     int skip = 0;
     int master_index = 0;
-    for (int i = 0; i < post_dofs; ++i) {
-      const int post_index = i;
+    int post_index = 0;
+    for (int i = 0; i < glued_dofs; ++i) {
       while (dof_is_slave[i + skip] && ((i + skip) < pre_dofs)) ++skip;
       const int pre_index = i + skip;
+      // skip boundary dofs
```
Comment (flx-wlf) [inline Bembel/src/AnsatzSpace/Glue.hpp:140]: "boundary" is ambiguous as we deal with BEM. Patch boundary?

File: Bembel/src/AnsatzSpace/Glue.hpp [hunk 3]
```diff
@@ -485,6 +495,13 @@ struct glue_identificationmaker_<Derived, DifferentialForm::DivConforming> {
           d.coef = coef;
           out.push_back(d);
         }
+      } else {
```
Comment (flx-wlf) [inline Bembel/src/AnsatzSpace/Glue.hpp:498]: Update comment in line 486

File: Bembel/src/AnsatzSpace/Glue.hpp [hunk 4]
```diff
@@ -485,6 +495,13 @@ struct glue_identificationmaker_<Derived, DifferentialForm::DivConforming> {
...
+        for (int i = 0; i < size_of_edge_dofs; ++i) {
+          GlueRoutines::dofIdentification d;
+          d.dofs.push_back(dofs_e1[i]);
+          d.coef = 0;
```
Comment (flx-wlf) [inline Bembel/src/AnsatzSpace/Glue.hpp:502]: I stumbled over this assignment, as it suggest that the value of d.coef matters. As I understand, this coef is not needed, as the corresponding entry is skipped and the coefficient does not matter. Is my understanding correct?

**Comment by mx-nlte** [comment]
Visualization of a Screen example looks physically accurate.
Bembel_screen

**Comment by flx-wlf** [comment]
What does the coloring of the elements represent? Lets be on the safe side. Without solving an equation, can you use the function evaluator to visualize each dof separately? You can plot these to a time-series (same filename ending with an increasing number) which you can step through in paraview. Thus, visualizing every dof individualy, we can be sure the discrete space is as desired.

**Pull request closed without merging**